From 7d028a00e96517a0c9a13de1321f47dbba6edd1d Mon Sep 17 00:00:00 2001
From: KaviyaKumaresan <Kaviya.Kumaresan@ltts.com>
Date: Fri, 2 Sep 2022 15:59:13 +0530
Subject: [PATCH] REFPLTB-1775: RPI-B not getting erouter0 IP

Reason for change: Resolving 'undefined symbol: Get_Device_Mode'
Test procedure: Tested in local build
Risks: low

Signed-off-by: KaviyaKumaresan <Kaviya.Kumaresan@ltts.com>
Change-Id: I489da1435c3905bb8872124c18cacb3f15e92cfe
---
 .../TR-181/board_sbapi/cosa_dhcpv6_apis.c       | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c b/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c
index 20cf7114..c32fabd2 100755
--- a/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c
@@ -7441,6 +7441,23 @@ static token_t sysevent_token_1;
 #define ULA_ROUTE_SET "/tmp/.ula_route_set"
 #define DEF_ULA_PREF_LEN 64
 
+#if (defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)) && !defined (WAN_FAILOVER_SUPPORTED)
+//For RPI & Turris platform, adding Get_Device_Mode implementation when only RDKB_EXTENDER_ENABLED is enabled.
+int Get_Device_Mode()
+{
+    int deviceMode = 0;
+    char buf[8]= {0};
+    memset(buf,0,sizeof(buf));
+    if ( 0 == syscfg_get(NULL, "Device_Mode", buf, sizeof(buf)))
+    {
+        if (buf[0] != '\0' && strlen(buf) != 0 )
+            deviceMode = atoi(buf);
+    }
+    return deviceMode;
+
+}
+#endif
+
 void enable_Ula_IPv6(char* ifname)
 {
     CcspTraceInfo(("%s : ENTRY\n",__FUNCTION__));
-- 
2.36.1

